quickwit/quickwit-opentelemetry/src/otlp/mod.rs

```diff
@@ -131,6 +133,62 @@ fn is_zero(count: &u32) -> bool {
     *count == 0
 }
 
+pub fn extract_otel_traces_index_patterns_from_metadata(
+    metadata: &tonic::metadata::MetadataMap,
+    otel_index_header_name: &str,
+) -> Result<Vec<String>, Status> {
+    let comma_separated_index_id_patterns = metadata
+        .get(otel_index_header_name)
+        .map(|index| index.to_str().map(|index| index.to_string()))
+        .transpose()
+        .map_err(|error| {
+            Status::internal(format!(
+                "Failed to extract index ID from request metadata: {}",
+                error
+            ))
+        })?
+        .unwrap_or_else(|| OTEL_TRACES_INDEX_ID_PATTERN.to_string());
+    let mut index_id_patterns = Vec::new();
+    for index_id_pattern in comma_separated_index_id_patterns.split(',') {
+        validate_index_id_pattern(index_id_pattern).map_err(|error| {
+            Status::internal(format!(
+                "Invalid index ID pattern in request metadata: {}",
+                error
+            ))
+        })?;
+        index_id_patterns.push(index_id_pattern.to_string());
+    }
+    Ok(index_id_patterns)
+}
+
+pub(crate) fn extract_otel_traces_index_from_metadata(
+    metadata: &tonic::metadata::MetadataMap,
+    otel_index_header_name: &str,
+) -> Result<String, Status> {
+    let index_id = metadata
+        .get(otel_index_header_name)
+        .map(
+            |index: &tonic::metadata::MetadataValue<tonic::metadata::Ascii>| {
```

Comment:
Here, I think performing the conversion to `String` later will make the code more readable.